### DISCUSSION TITLE
broken text object in tsx files

### DISCUSSION THREAD

**Discussion by cStralpt** [LazyVim/LazyVim#1341] [Q&A] [answered]
how to fix this issues?, it's annoying,
vat/vit
![nvim bugs](https://github.com/LazyVim/LazyVim/assets/95400822/82381fc7-17ca-46f2-9d5c-a55a7405cb0a)

**Comment by dpetka2001** [comment]
Can you try with `vaf/vif`? Does it behave how you want?

**Reply by cStralpt** [reply]
vat/vit is to select anything inside/around tags, thats what i want, but it just doing a half, even it select the outer parent tag with the vat, it's so weird

**Reply by dpetka2001** [reply]
I know. Did you try `vaf/vif` to see if it works for you?

**Reply by dpetka2001** [reply]
This is a known limitation of `mini.ai` plugin. See this [issue](https://github.com/echasnovski/mini.nvim/issues/156). It forwards to another issue, which has this [answer](https://github.com/echasnovski/mini.nvim/issues/110#issuecomment-1212277863) as solution. Lazyvim already incorporates this solution in `vaf/vif`, so I believe it should work for you. If instead you would like to use `vat/vit` nonetheless, you should do what that answer suggests. Or if it's not working, then you should open an issue at `mini.nvim` repo, as that's the plugin that's responsible for this kind of behavior.

**Reply by cStralpt** [reply]
it working fine, but it just do selecting a function, i hope on the latest update(min.nvim) it fix the problem

**Reply by dpetka2001** [reply]
As the maintainer has explicitly stated it [won't be resolved](https://github.com/echasnovski/mini.nvim/issues/420#issuecomment-1636731426) 
> 'mini.ai' uses Lua patterns to search for the target region. This will inevitably result into false positives. It is a [known issue which won't be resolved'](https://github.com/echasnovski/mini.nvim/blob/2b37c5e1d2a3f62ba4060a819f17d8d976018928/doc/mini-ai.txt#L43-L45).
Tags is a main source of such false positives. In particular, I couldn't find a way to make it work with nested similar tags, like `<div>`s in your example.

Have you tried the solution in this [comment](https://github.com/echasnovski/mini.nvim/issues/110#issuecomment-1212277863). Just create a file in `~/.config/nvim/after/ftplugin/html.lua` and copy/paste the code snippet that is provided in the comment. It's as simple as that.

**Reply by cStralpt** [reply]
it tried to put the recommended way, by putting the file to this directory ```after/ftplugin/html.lua``` and ```lua/plugins/html.lua```
but both doesn't work for me.

**Reply by dpetka2001** [reply]
Then I would recommend you to make an issue at `mini.nvim` repo regarding the problem you're facing. Its maintainer will surely be able to give you a more appropriate solution, since he knows better how the plugin works.

**Reply by cStralpt** [reply, marked as answer]
update: to fix this issues just add this to your config `./lua/plugins/mini-ai.lua`:
 ```
 return  {
      "echasnovski/mini.ai",
      opts = function(_, opts)
        opts.custom_textobjects["t"] = false
      end,
    }
```
while this PR is still open [https://github.com/LazyVim/LazyVim/pull/1434#issue-1889078376](url)

**Reply by abeldekat** [reply]
Hello @cStralpt,

Somehow, the link to the PR is not working.
The PR can be found [here](https://github.com/LazyVim/LazyVim/pull/1434)
